package.json

```diff
@@ -3,12 +3,13 @@
   "license": "LGPL-3.0-or-later",
   "scripts": {
     "clean": "rm -rf node_modules yarn-error.log css static/css *.log *.sqlite _site/ build/ .jekyll-metadata .bundle",
-    "watch": "watch -p 'static/sass/**/*' -p 'static/js/**/*.js' -c 'yarn run build'",
+    "watch": "watch -p 'static/sass/**/*' -p 'static/js/**/*.js' -p 'static/js/**/*.ts' -c 'yarn run build'",
```

Comment:
What's the difference between a `.js` and a `.ts` file? Can you simply take a plain `.js` file, whack `.ts` on the end and compile it with typescript, without any other changes?

Because if so, I'd suggest we just rename them all now and compile them all with typescript, and rip out `.js` support altogether.

Reply:
There are some minor changes that may be required but I'd prefer we don't do what you're proposing as that's the bare minimum. I'd prefer that when something gets converted to TS we have made the conscious effort to go through the file and truly "convert" it.

Reply:
If we go with the approach of strict TypeScript rules, then you cannot just change `*.js` to `*.ts` because it will fail. If you change the file to `*.ts` you will have to deal with all the errors and amend the file so it passes the type check.

I think we could do that, and then incrementally make the rules stricter, however I don't like that approach as you will have to fix all the files when you change the strictness. Also it will create inconsistencies between projects, as some will have stricter rules than others overtime.

Reply:
Ok thanks for explaining.